Flexibility in "name" of request type (#428)

benrfairless opened the pull request:
This addresses #428 by using the tags in [public_body_categories_en.rb](https://github.com/openaustralia/righttoknow/blob/production/lib/public_body_categories_en.rb) to determine:
- The request type (e.g. `Government Information (Public Access)` for NSW, or `Right to Information` for QLD and TAS)
- The appropriate acronym for the request type (e.g. `GIPA` for NSW, or `RTI` for QLD and TAS)
- The relevant Act of Parliament used (e.g. `Government Information (Public Access) Act 2009 (NSW)`  for NSW
- The request type in part of a sentence (e.g. `A Right to Information request` for TAS)

Anywhere the relevant defs are used in Alaveteli they should now reflect the name for the relevant jurisdiction.

Things to be aware of:
- This requires that the tags used are case sensitive. `NSW_state` will work, but `nsw_state` and `NSW_STATE` won't.

before @henare asks (:wink:), I've tested this in vagrant (making requests) and can see no major issues. I'm happy for it to go into staging before production if you would prefer.

benrfairless committed:
Delete OutgoingMessage changes as we don't use these

benrfairless committed:
Define the ACT in the PublicBody Module

benrfairless committed:
Define NSW in the PublicBody Module

benrfairless committed:
Define NT in the PublicBody Module

benrfairless committed:
Define QLD in the PublicBody Module

benrfairless committed:
Define SA in the PublicBody Module

benrfairless committed:
Define TAS in the PublicBody Module

benrfairless committed:
Define VIC in the PublicBody Module

benrfairless committed:
Define WA in the PublicBody Module

benrfairless committed:
Define Federal Authorities in the PublicBody Module

benrfairless committed:
Minor change to comment for readability

benrfairless committed:
Copied the definitions as they are from https://github.com/openaustralia/alaveteli/blob/production/app/models/info_request.rb#L301-L337

benrfairless committed:
Define the law for ACT

benrfairless committed:
Add comments to identify ACT/Fallback

benrfairless committed:
Define the law for NSW

benrfairless committed:
More Comments to make it look tidy

benrfairless committed:
Define the law for NT

benrfairless committed:
Fix NSW Law name

benrfairless committed:
Fix NT Law name

benrfairless committed:
Define the law for QLD

benrfairless committed:
Define the law for SA

benrfairless committed:
Define the law for TAS

benrfairless committed:
QLD Typo

benrfairless committed:
Define the law for VIC

benrfairless committed:
Define the law for WA

benrfairless committed:
Add additional "end" on model_patches.rb

benrfairless committed:
Typo change to reflect correct state

henare commented:
This is a great start @brfairless!

It's a bit verbose though. Since councils come under state jurisdiction I don't think we need to separate them out in the code where possible.

There's also commonality with the code [I've been developing](https://github.com/openaustralia/righttoknow/issues/73#issuecomment-78418553) to adjust response timeframes. Since that can't be merged until we upgrade, which is also waiting for the responsive design changes to be completed, I'll make the changes to this pull request and see if we can deploy this.

henare committed:
Use implicit returns

henare committed:
Don't use `self.` because it's unnecessary

henare committed:
Fix indentation and make it consistent only

henare committed:
Replace separate jurisdiction methods with a single one

henare committed:
Create method like #law_used that returns the applicable law

We can't just override #law_used since it has validations on it and it
feels cleaner not to override a database backed attribute in our model
patches.

henare committed:
Simplify methods by just checking #australian_law_used

henare committed:
Remove comment

henare committed:
Remove EIR since it's not a law used in Australia

henare committed:
Make conditionals more concise by using case statements

henare committed:
Use interpolation

henare committed:
Add some comments to explain where these methods are used

henare commented:
OK I think this is ready to go.

I've been reminded of why I wasn't more excited when I looked at this approach a while back - it only changes the name on a few places for the requester and on the outbound mail to the authority.

This is OK but it doesn't change the name of requests anywhere else like on the public body page. That means it's inconsistently displayed to the requester which is a bit unpleasant.

Anyway, it's a good start and it does the bit we really needed to cover which is the outbound email to authorities.
